cheatsheets/Server_Side_Request_Forgery_Prevention_Cheat_Sheet.md

```diff
@@ -0,0 +1,394 @@
+
+The Firewall component, as a specific device or using the one provided within the operating system, will be used here to define the legitimate flows. 
+
+In the schema below, a Firewall component is leveraged to limit the application's access, and in turn, limit the impact of an application vulnerable to SSRF:
+
+![Case 1 for Network layer protection about flows that we want to prevent](../assets/Server_Side_Request_Forgery_Prevention_Cheat_Sheet_Case1_NetworkLayer_PreventFlow.png)
+
+[Network segregation](https://www.mwrinfosecurity.com/our-thinking/making-the-case-for-network-segregation) (see this set of [implementation advices](https://www.cyber.gov.au/publications/network-segmentation-and-segregation)) can also be leveraged and **is highly recommanded in order to block illegitimate calls directly at network level itself**.
+
+## Case 2 - Application can send requests to ANY external IP address or domain name
+
+This case happens when user can control an URL to an **External** resource and the application makes a request to this URL (e.g. in case of [WebHooks](https://en.wikipedia.org/wiki/Webhook)). Whitelist cannot be used here because the list of IPs/domains is often unknown upfront and is dynamically changing. 
+
+In this scenario, *External* refers to any IP that is doesn't belong to the internal network, and should be reached by going over the public internet.
+
+Thus, the call from the *Vulnerable Application*: 
+* **Is NOT** targeting one of the IP/domain *located inside* the company's global network.
+* Uses a convention defined between the *VulnerableApplication* and the expected IP/domain in order to *prove* that the call has been legitimately initiated.
+
+### Challenges in blocking URLs at application layer
+
+Based on the business requirements of the above mentioned applications, the whitelist approach is not a valid solution. Despite knowing that the blacklist approach is not an impenetrable wall, it is the best solution in this scenario. It helps tell the application what it should **not** do.
+
+Here is why filtering URLs is hard at the Application layer:
+
+* It implies that the application must be able to detect, at the code level, that the provided IP (V4 + V6) is not part of the official [private networks ranges](https://en.wikipedia.org/wiki/Private_network) including also *localhost* and *IPv4/v6 Link-Local* addresses. Not every SDK provides a built-in feature for this kind of verification, and leaves the handling up to the developer understand all of its pitfalls and possible values, which makes it a demanding task.
+* Same remark for domain name: The company must maintains a list of all internal domain names and provide a centralized service to allow an application to verify if a provided domain name is an internal one. For this verification, an internal DNS resolver can be queried by the application but this internal DNS resolver must not resolve external domain names.
```

Comment:
remove the `s` from maintains: The company must maintain a list ...

Reply:
Taken in account in the coming commit.